plugins/modules/user_role_workflow_manager.py

```diff
@@ -2975,6 +3083,64 @@ def verify_diff_deleted(self, config):
 
         return self
 
+    def update_user_role_profile_messages(self):
+        """
+        Updates and logs messages based on the status of users and roles.
+        Args:
+            self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+        Returns:
+            self (object): Returns the current instance of the class with updated `result` and `msg` attributes.
+        Description:
+            This method aggregates status messages related to the creation, update, or deletion of users and roles.
+            It checks various instance variables (`create_user`, `update_user`, `no_update_user`, `delete_user`,
+            `create_role`, `update_role`, `no_update_role`, `delete_role`) to determine the status and generates
+            corresponding messages. The method also updates the `result["response"]` attribute with the concatenated status messages.
+        """
+
+        self.result["changed"] = False
+        result_msg_list = []
+
+        if self.created_user:
+            create_user_msg = "User(s) '{0}' created successfully in Cisco Catalyst Center.".format("', '".join(self.created_user))
+            result_msg_list.append(create_user_msg)
+
+        if self.updated_user:
+            update_user_msg = "User(s) '{0}' updated successfully in Cisco Catalyst Center.".format("', '".join(self.updated_user))
+            result_msg_list.append(update_user_msg)
+
+        if self.no_update_user:
+            no_update_user_msg = "User(s) '{0}' need no update in Cisco Catalyst Center.".format("', '".join(self.no_update_user))
+            result_msg_list.append(no_update_user_msg)
+
+        if self.deleted_user:
+            delete_user_msg = "User(s) '{0}' deleted successfully from the Cisco Catalyst Center.".format("', '".join(self.deleted_user))
+            result_msg_list.append(delete_user_msg)
+
+        if self.created_role:
+            create_role_msg = "Role(s) '{0}' created successfully in Cisco Catalyst Center.".format("', '".join(self.created_role))
+            result_msg_list.append(create_role_msg)
+
+        if self.updated_role:
+            update_role_msg = "Role(s) '{0}' updated successfully in Cisco Catalyst Center.".format("', '".join(self.updated_role))
+            result_msg_list.append(update_role_msg)
+
+        if self.no_update_role:
+            no_update_role_msg = "Role(s) '{0}' need no update in Cisco Catalyst Center.".format("', '".join(self.no_update_role))
+            result_msg_list.append(no_update_role_msg)
+
+        if self.deleted_role:
+            delete_role_msg = "Role(s) '{0}' deleted successfully from the Cisco Catalyst Center.".format("', '".join(self.deleted_role))
+            result_msg_list.append(delete_role_msg)
+
+        if self.created_user or self.updated_user or self.deleted_user or self.created_role or self.updated_role or self.deleted_role:
```

Comment:
Instead, we can simply check with result_msg_list and set the result.. 

(ie)
```
    if result_msg_list:
        self.result["changed"] = True
        self.msg = " ".join(result_msg_list)
    else:
        self.msg = "No changes were made. No user or role actions were performed in Cisco Catalyst Center."

    self.log(self.msg, "INFO")
    self.result["response"] = self.msg
    
     return self
```

Reply:
Updated the code accordingly.